File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int fibo[100001];
int m = 1234567;
int solution(int n) {
    int answer = 0;
    int ssize = n;
    fibo[0] = 0;
    fibo[1] = 1;
    for(int i=2; i<=ssize; i++){
        fibo[i] = (fibo[i-1] + fibo[i-2]) %m;
    }
    answer = fibo[ssize];
    return answer;
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int m = 1234567;
        int[] fibo = new int[100001];
        fibo[0] = 0;
        fibo[1] = 1;
        for(int i=2; i<=n; i++){
            fibo[i] = (fibo[i-1] + fibo[i-2]) % m;
        }
        answer = fibo[n];
        return answer;
    }
}